### ISSUE TITLE
How do you namespace migrations?

### ISSUE THREAD

**Issue by Bilge** [cakephp/phinx#1820]
The [template](https://github.com/cakephp/phinx/blob/ac22eb22bd6bc36fab58610f385bc1f12d7f2150/src/Phinx/Migration/Migration.template.php.dist) references `$namespaceDefinition`, which according to the code comes from some config, somewhere. But the [config documentation](https://book.cakephp.org/phinx/0/en/configuration.html) makes no reference to namespaces anywhere, except [aliases](https://book.cakephp.org/phinx/0/en/configuration.html#aliases). How does one actually set the migration namespace with config?

**Comment by MasterOdin** [comment]
You would do:

```php
    'paths' => [
        'migrations' => [
            'Foo\\Test' => '%%PHINX_CONFIG_DIR%%/db/migrations'
        ],
        'seeds' => [
            'Foo\\Bar' => '%%PHINX_CONFIG_DIR%%/db/seeds'
        ]
    ],
```

where the key is the namespace and value is the path.

**Comment by Bilge** [comment]
Is it documented?

**Comment by MasterOdin** [comment]
No, and I agree it should be. The above is as much a note to myself as anyone else who wants to update the docs as it is an immediate answer to your question on how to do it.

**Comment by jcapcik** [comment]
I was wondering how to namespace migrations as well, but it looks as though this is already documented in https://github.com/cakephp/phinx/blob/master/docs/en/namespaces.rst, which isn't included in the `contents`

**Comment by dereuromark** [comment]
In Migrations we now go with anonymous classes (as other frameworks did too in the past):
https://github.com/cakephp/migrations/pull/882

That would resolve the namespace issue, too, since it now only cares about the filename (which it already does for the timestamp anyway).
